library/std/src/collections/hash/map.rs

```diff
@@ -880,7 +880,8 @@ where
         self.base.get(k)
     }
 
-    /// Returns the key-value pair corresponding to the supplied key.
+    /// Returns the key-value pair corresponding to the supplied key. This is
```

Comment:
When you index a hashmap, you don't actually use `K` for indexing, but you use `Q` that upholds `K: Borrow<Q>`. So another use-case could be that you want to inspect `&K` when you only have `&Q`, right?

Reply:
Yeah, I think the `K`/`Q` difference would be more interesting to demonstrate.

Reply:
Hmm, let me explain how this PR came about. This method has one genuine use in the compiler, in `rustc_resolve`. (There are three call sites which are all extremely similar.) I had never heard of this method, so I looked up the docs and found them unhelpful. After more digging I discovered that the method was being used in the way I described/demonstrated in this PR: with a type where one of the fields was being ignored.

So while I can see that the `&K`/`&Q` case is possible, the very limited real-world experience I have suggests the "ignored field" case is more interesting. "you want to inspect e.g. the capacity of the `&String` key" feels not that useful.

Also, I just looked through the version control history. The `get_key_value` methods were added in #49346, implementing the suggestion from #43143. The only justification there is "sometimes you need a reference to a key with the lifetime of the collection", which is a third motivation, but there are no actual examples and surprisingly little discussion :(

So now my inclination is to expand the text description of the methods to include the `&K`/`&Q` case and the "reference to a key with the lifetime of the collection" case, but keep the examples on the "ignore a field" case. Thoughts?

Reply:
That sounds good to me. I agree that your use-case is probably the most useful one.